Replace bcrypt with bcryptjs for better cross-platform compatibility

## Problem

JavaScriptSolidServer currently uses `bcrypt` as the primary password hashing library, with `bcryptjs` as a fallback. However, `bcrypt` is a native C++ module that requires compilation, which causes installation failures on several platforms:

### Platforms Affected
- ✅ **Termux/Android** - Fails with `gyp: Undefined variable android_ndk_path`
- ✅ **Windows** - Common compilation errors with node-gyp
- ✅ **ARM devices** - Various build issues
- ✅ **Restricted environments** - No build tools available

### Current Behavior
When users try to install JavaScriptSolidServer (or wrappers like jspod/fastpod) on these platforms:

```
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! code 1
```

This completely blocks installation, even though `bcryptjs` is already in the dependencies.

## Proposed Solution

**Switch to bcryptjs as the primary/only password hashing library.**

### Why This Makes Sense

1. **Minimal Performance Impact**
   - bcrypt (C++): ~100ms per hash
   - bcryptjs (JS): ~130ms per hash
   - **Difference: 30ms** - only during login/registration (rare operations)

2. **Better User Experience**
   - No compilation errors
   - Works everywhere Node.js runs
   - Faster npm install (no build step)
   - Same security guarantees

3. **Real-World Usage**
   - Password hashing happens maybe 1-10 times per day per user
   - File operations happen thousands of times per day
   - The 30ms difference is completely negligible

4. **Already a Dependency**
   - bcryptjs is already in package.json
   - No new dependencies needed
   - Just remove bcrypt

### Security

Both libraries implement the same bcrypt algorithm with the same security properties:
- ✅ Salt generation
- ✅ Configurable work factor
- ✅ Resistance to timing attacks
- ✅ Industry-standard algorithm

The only difference is implementation language (C++ vs JavaScript).

## Testing

Tested on:
- ✅ Termux/Android (ARM64, Android 15) - **Currently fails with bcrypt**
- ✅ Linux (works with both)
- ✅ Expected to fix Windows issues as well

## Implementation

Change from:
```json
"dependencies": {
  "bcrypt": "^6.0.0",
  "bcryptjs": "^3.0.3"
}
```

To:
```json
"dependencies": {
  "bcryptjs": "^3.0.3"
}
```

And update all `require('bcrypt')` to `require('bcryptjs')`.

## Impact

This would enable:
- 📱 Running Solid pods on Android/Termux
- 💻 Easier installation on Windows
- 🌍 Better cross-platform support overall
- ⚡ Faster installation (no compilation)

With virtually zero downside (30ms during infrequent auth operations).

---

**Use Case:** Trying to run `npx fastpod` on Termux fails due to bcrypt compilation. This blocks the entire "just works" philosophy of having a zero-config Solid server.

--- a/README.md
+++ b/README.md
@@ -60,7 +60,7 @@ A minimal, fast, JSON-LD native Solid server.
 
 ### Android/Termux
 
-JSS runs on Android via Termux with automatic `bcryptjs` fallback:
+JSS runs on Android via Termux (uses pure JavaScript `bcryptjs` for compatibility):
 
 ```bash
 pkg install nodejs git
@@ -1045,7 +1045,7 @@ Minimal dependencies for a fast, secure server:
 - **jose** - JWT/JWK handling for Solid-OIDC
 - **n3** - Turtle parsing (only used when conneg enabled)
 - **oidc-provider** - OpenID Connect Identity Provider (only when IdP enabled)
-- **bcrypt** - Password hashing (only when IdP enabled)
+- **bcryptjs** - Password hashing (only when IdP enabled)
 - **microfed** - ActivityPub primitives (only when activitypub enabled)
 - **better-sqlite3** - SQLite storage for federation data
 

--- a/package.json
+++ b/package.json
@@ -27,7 +27,6 @@
     "@fastify/rate-limit": "^9.1.0",
     "@fastify/websocket": "^8.3.1",
     "@simplewebauthn/server": "^13.2.2",
-    "bcrypt": "^6.0.0",
     "bcryptjs": "^3.0.3",
     "better-sqlite3": "^12.5.0",
     "commander": "^14.0.2",

--- a/src/idp/accounts.js
+++ b/src/idp/accounts.js
@@ -4,13 +4,8 @@
  * Email is optional - internally uses username@jss if not provided
  */
 
-// Try native bcrypt, fall back to pure JS bcryptjs (for Android/Termux)
-let bcrypt;
-try {
-  bcrypt = await import('bcrypt').then(m => m.default);
-} catch {
-  bcrypt = await import('bcryptjs').then(m => m.default);
-}
+// Use bcryptjs for cross-platform compatibility (works on Android/Termux/Windows)
+const bcrypt = await import('bcryptjs').then(m => m.default);
 import crypto from 'crypto';
 import fs from 'fs-extra';
 import path from 'path';

